src/explore-education-statistics-admin/src/components/NavigableSections.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { ReactNode } from 'react';
+import PreviousNextLinks from '@admin/components/PreviousNextLinks';
+import Link from '@admin/components/Link';
+
+export interface Section<SectionType extends {}> {
+  section: SectionType;
+  label: string;
+  linkTo: string;
+}
+
+interface Props<SectionType> {
+  navigationHeadingText: string;
+  navigationHeadingSubtitle: string;
+  availableSections: Section<SectionType>[];
+  selectedSection: Section<SectionType>;
+  children: ReactNode;
+}
+
+/**
+ * Represents a portion of a page with navigable tabbed sections, and Next / Previous buttons to switch between the
+ * sections.  A client of this component provides a full ordered list of the available sections, and the currently
+ * selected section.  From this, the Previous and Next links are inferred.
+ *
+ * The client also provides the appropriate body for the selected section, which is rendered within this component
+ * as its props.children.
+ *
+ * The SectionType generic type represents a type for a unique id per section, whatever works best for the client.
+ * This could for instance be an enum of available sections, or a string or number id.
+ *
+ * @param availableSections
+ * @param selectedSection
+ * @param children
+ * @constructor
+ */
+const NavigableSections = <SectionType extends {}>({
+  availableSections,
+  selectedSection,
+  children,
+}: Props<SectionType>) => {
+  const nextSection =
+    availableSections.indexOf(selectedSection) < availableSections.length - 1
+      ? availableSections[availableSections.indexOf(selectedSection) + 1]
+      : null;
+
+  const previousSection =
+    availableSections.indexOf(selectedSection) > 0
+      ? availableSections[availableSections.indexOf(selectedSection) - 1]
+      : null;
+
+  return (
+    <>
+      <MainSectionNavigation
+        selectedSection={selectedSection.section}
+        availableSections={availableSections}
+      />
+      {children}
+      <PreviousNextLinks
+        previousSection={previousSection || undefined}
+        nextSection={nextSection || undefined}
+      />
+    </>
+  );
+};
+
+interface NavigationProps<SectionType> {
+  availableSections: Section<SectionType>[];
+  selectedSection: SectionType;
+}
+
+const MainSectionNavigation = <SectionType extends {}>({
```

Comment:
Inline inline inline :wink:

Reply:
Ahh, it just looks really bloated and unclear when it's all bundled in together is all.  I prefer it to be a bit clearer.

Reply:
Fair enough, it is a bit of a taste thing as I don't really think this component is particularly complicated enough to warrant splitting out. To me `MainSectionNavigation` is actually the meat and potatoes of `NavigableSections`!

Reply:
I ended up rewriting a lot, and most is now inlined, thanks